spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/cassandra/CassandraMetricsAutoConfigurationTests.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.metrics.cassandra;
+
+import java.util.Arrays;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.actuate.autoconfigure.metrics.test.MetricsRun;
+import org.springframework.boot.actuate.metrics.cassandra.CassandraMetricsBinder;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration;
+import org.springframework.boot.autoconfigure.cassandra.DriverConfigLoaderBuilderCustomizer;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link CassandraMetricsAutoConfiguration}.
+ *
+ * @author Erik Merkle
+ */
+public class CassandraMetricsAutoConfigurationTests {
```

Comment:
These tests will only work if there is a Cassandra node listening on localhost:9042. You need to mock the session instead:

```
public class CassandraMetricsAutoConfigurationTests {

	private static final String METRIC_NAME = CassandraMetricsBinder.SESSION_PREFIX + ".s0.bytes-sent";

	private ApplicationContextRunner contextRunner = new ApplicationContextRunner().with(MetricsRun.simple())
			.withConfiguration(
					AutoConfigurations.of(CassandraAutoConfiguration.class, CassandraMetricsAutoConfiguration.class));

	/**
	 * Cassandra driver metrics should be enabled by default as long as the desired
	 * metrics are enabled in the Driver's configuration.
	 */
	@Test
	void autoConfiguredCassandraIsInstrumented() {
		this.contextRunner
				.withBean(CqlSession.class, () -> mockSession(true))
				.run((context) -> {
					MeterRegistry registry = context.getBean(MeterRegistry.class);
					Meter meter = registry.find(METRIC_NAME).meter();
					assertThat(meter).isNotNull();
					assertThat(meter.measure()).hasSize(1).extracting("value").containsExactly(42d);
				});
	}

	/**
	 * Cassandra driver metrics should be disabled if set to false via Spring metrics
	 * management.
	 */
	@Test
	void cassandraInstrumentationCanBeDisabledThroughSpringMetricsManagement() {
		this.contextRunner
				.withBean(CqlSession.class, () -> mockSession(true))
				.withPropertyValues("management.metrics.enable.cassandra.driver.session=false")
				.run((context) -> {
					MeterRegistry registry = context.getBean(MeterRegistry.class);
					assertThat(registry.find(METRIC_NAME).meter()).isNull();
				});
	}

	/**
	 * Cassandra driver metrics should be disabled if no metrics are enabled in the
	 * Driver's configuration.
	 */
	@Test
	void cassandraInstrumentationIsDisabledIfNoMetricsEnabledInDriverConfig() {
		this.contextRunner
				.withBean(CqlSession.class, () -> mockSession(false))
				.run((context) -> {
					MeterRegistry registry = context.getBean(MeterRegistry.class);
					assertThat(registry.find(METRIC_NAME).meter()).isNull();
				});
	}

	private CqlSession mockSession(boolean enableMetrics) {
		CqlSession session = mock(CqlSession.class);
		given(session.getName()).willReturn("s0");
		if (enableMetrics) {
			Metrics metrics = mock(Metrics.class);
			given(session.getMetrics()).willReturn(Optional.of(metrics));
			MetricRegistry metricRegistry = new MetricRegistry();
			given(metrics.getRegistry()).willReturn(metricRegistry);
			metricRegistry.meter("s0.bytes-sent").mark(42);
		}
		else {
			given(session.getMetrics()).willReturn(Optional.empty());
		}
		return session;
	}
}
```

Reply:
You're right, I missed this when testing and didn't realize I had a CCM instance up.